serviceaccounts.rb

```diff
@@ -35,33 +39,29 @@ def existing_file_account(path)
   def run()
     common = Common.new
     ENV["GOOGLE_APPLICATION_CREDENTIALS"] = @path
-    service_account = @service_account
-    if not service_account
-      service_account = "#{@project}@appspot.gserviceaccount.com"
-    end
-    if service_account == existing_file_account(@path)
+    if @service_account == existing_file_account(@path)
       # Don't generate another key if this account is already active. This can
       # happen for nested service account contexts, for example.
       common.status "Attaching to existing keyfile @ #{@path}"
       yield
       return
     end
 
-    if service_account == "all-of-us-workbench-test@appspot.gserviceaccount.com"
+    if @service_account == "all-of-us-workbench-test@appspot.gserviceaccount.com"
       unless File.exists?(@path)
         common.run_inline %W{gsutil cp gs://#{@project}-credentials/app-engine-default-sa.json
             #{@path}}
       end
       yield
```

Comment:
I was actually just using this yesterday and I was wondering if it would be helpful if this yielded some kind of context struct about the project, service account, and maybe the path.

Reply:
WDYT about `yield(self)` here? I can add readers for the other attributes you mentioned if desired

Reply:
Interesting. That would work. If it's just reader attributes then it would be safer too.